src/backend/schemas/metrics.py

```diff
@@ -1,17 +1,38 @@
 import time
 import uuid
+from enum import Enum
 from typing import Any
 
 from pydantic import BaseModel
 
 
+class MetricsMessageType(str, Enum):
+    # implemented
+    USER_CREATED = "user_created"
+    USER_UPDATED = "user_updated"
+    USER_DELETED = "user_deleted"
+    ASSISTANT_CREATED = "assistant_created"
+    ASSISTANT_UPDATED = "assistant_updated"
+    ASSISTANT_DELETED = "assistant_deleted"
+    ASSISTANT_ACCESSED = "assistant_accessed"
+    CHAT_API_SUCCESS = "chat_api_call_success"
+    CHAT_API_FAIL = "chat_api_call_failure"
+    RERANK_API_SUCCESS = "rerank_api_call_success"
+    RERANK_API_FAIL = "rerank_api_call_failure"
+    # pending
+    ENV_LIVENESS = "env_liveness"
+    COMPASS_NEW_INDEX = "compass_new_index"
+    COMPASS_REMOVE_INDEX = "compass_remove_index"
+    COMPASS_NEW_USER = "compass_new_user"
+    COMPASS_REMOVE_USER = "compass_remove_user"
+    UNKNOWN_SIGNAL = "unknown"
+
+
 class MetricsDataBase(BaseModel):
     id: str
     user_id: str
     trace_id: str
-    method: str
-    endpoint_name: str
-    success: bool
```

Comment:
do the new message type encapsulate the success status? for example, i see we have success and fail types but for something like `assistant_created` there is no indication that it succeeded or failed, or is that not important for the metrics use case?

Reply:
The main reason we sent success before was for the mapping to event name. Without that, the success isn't needed. The goal long term would be to remove the mapping altogether and simply work with event types for the events we care about.